File: Day 24 : More Linked Lists/Solution.rb
class Node
  attr_accessor :data,:next

  def initialize data
    @data = data
    @next = nil
  end
end

class Solution

  def removeDuplicates(head)
    if head == nil
      return head
    else
      root = head
      while root.next != nil
        if root.data == root.next.data
          root.next = root.next.next
        else
          root = root.next
        end
      end
    end
    return head
  end

  def insert(head,value)
      p=Node.new(value)
      if head==nil
        head=p
      elsif head.next==nil
        head.next=p
      else  
        current = head
        while current.next != nil
          current = current.next
        end 
        current.next = p
      end
    return head   
  end 	

  def display(head)
    current = head
    while current 
      print current.data," "
      current = current.next
    end
  end
end
     
mylist = Solution.new
head = nil
T = gets.to_i
for i in 1..T
  data = gets.to_i
  head = mylist.insert(head,data)
end
head = mylist.removeDuplicates(head)
mylist.display(head)

File: Day 24 : More Linked Lists/Solution.ts
class MyNode {
    data: number
    next: MyNode | null

    constructor(data: number, next: MyNode | null) {
        this.data = data;
        this.next = next;
    }
}

class Solution {
    removeDuplicates(head: MyNode | null): MyNode | null {
        if (head == null) {
            return head;
        } else {
            let root: MyNode = head;
            while (root.next) {
                if (root.data == root.next.data) {
                    root.next = root.next.next;
                } else {
                    root = root.next;
                }
            }
        }

        return head;
    }

    insert(head: MyNode | null, data: number): MyNode | null {
        let node = new MyNode(data, null);
        node.data = data;

        if (head == null) {
            head = node;
        } else {
            let current = head;
            while (current.next) {
                current = current.next;
            }
            current.next = node;
        }

        return head;
    }

    display(head: MyNode | null): void {
        let start: MyNode | null = head;
        while (start) {
            process.stdout.write(start.data + " ");
            start = start.next;
        }
    }
}

function main() {
    const T: number = +readLine();
    let head: MyNode | null = null;
    let mylist: Solution = new Solution();
    for (let i: number = 0; i < T; i++) {
        let data: number = +readLine();
        head = mylist.insert(head, data);
    }
    head = mylist.removeDuplicates(head);
    mylist.display(head);
}